build.gradle

```diff
@@ -76,6 +76,22 @@ subprojects {
     }
 
     project.configurations.all {
+        // Dependencies can't depend on a different major version of Glean than A-C itself.
+        resolutionStrategy.eachDependency { details ->
+            if (details.requested.group == 'org.mozilla.telemetry'
+                    && details.requested.name.contains('glean') ) {
+                def requested = details.requested.version.tokenize(".")
+                def defined = Versions.mozilla_glean.tokenize(".")
+                // Check the major version
+                if (requested[0] != defined[0]) {
+                    throw new AssertionError("Cannot resolve to a single Glean version. Requested: ${details.requested.version}, A-C uses: ${Versions.mozilla_glean}")
```

Comment:
Drive-by nit: could we also print `details.requested.name` in the error so if this happens in CI, we can see which dep has the incorrect version? Not certain if this is valuable or if the error trace will include the offending dependency already.

Reply:
Ah, the docs around this are ... bad, so I didn't find the `requested.name`. I'll include it.

Reply:
Nevermind, I misunderstood.
`details.requested.name` will always be `glean` (the `if` just above limits this).
We unfortunately do not get the info which other dependency requested this Glean version.